### ISSUE TITLE
OrdinalEncoder issue in GridSearchCV

### ISSUE THREAD

**Issue by karlenander** [snowflakedb/snowflake-ml-python#43]
Hi!
executing code similar to below results in an error
```python
 pipeline = Pipeline(
        steps=[
            (
                "preprocessing",
                ColumnTransformer(
                    transformers=[
                        (
                            "ORD",
                            OrdinalEncoder(handle_unknown="use_encoded_value", unknown_value=np.nan),
                            categorical_features,
                        ),
                        ("MMS", MinMaxScaler(clip=True), numerical_features),
                    ]
                ),
            ),
            ("REG", XGBRegressor(label_cols=[target_name])),
        ]
    )

    parameters = {
        "REG__max_depth": [5, 10,],
        "REG__n_estimators": [50],
        "REG__learning_rate": [0.1],
    }
    grid = GridSearchCV(
        estimator=pipeline,
        param_grid=parameters,
        input_cols=categorical_features + numerical_features,
        label_cols=[target_name],
        ...

    )
    grid.fit(train_df)
```
error message
```bash
line 141, in get_filtered_valid_sklearn_args
        if isinstance(val, float) and (np.isnan(val) and np.isnan(default_sklearn_args[key])):
    TypeError: ufunc 'isnan' not supported for the input types, and the inputs could not be safely coerced to any supported types according to the casting rule ''safe''
```
When using unknown_value=100 or similar, everything works as expected (only have issue when setting unknown_value=np.nan)

**Comment by sfc-gh-xjiang** [comment]
Hi @karlenander, this bug is currently fixed and released in version of 1.0.8. Thanks for reporting the issue, and please try with our latest package to see if the issue persists.

**Comment by karlenander** [comment]
works! thanks for fixing.
